Library/ViewModels/DashboardViewModel.swift

```diff
@@ -32,6 +32,12 @@ public protocol DashboardViewModelInputs {
   /// Call when the projects drawer has animated out.
   func dashboardProjectsDrawerDidAnimateOut()
 
+  /// Call to open message thread for specific project
+  func messageThreadNavigated(projectId: Param, messageThread: MessageThread)
+
+  /// Call to open project messages thread
+  func openMessageThreadRequested()
```

Comment:
May be.. I try to treat view model inputs similarly to Command Pattern - You don't know who triggered the command you only know what the command should do. But may be my approach is incorrect? We can chat briefly about it.